src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -128,32 +128,38 @@ pub struct WorkunitMetadata {
   pub stdout: Option<hashing::Digest>,
   pub stderr: Option<hashing::Digest>,
   pub artifacts: Vec<(String, hashing::Digest)>,
+  pub user_metadata: Vec<(String, UserMetadataItem)>,
 }
 
 impl WorkunitMetadata {
   pub fn new() -> Self {
-    WorkunitMetadata::default()
-  }
-
-  pub fn with_level(level: Level) -> Self {
-    let mut metadata = WorkunitMetadata::default();
-    metadata.level = level;
-    metadata
-  }
-}
-
-impl Default for WorkunitMetadata {
```

Comment:
Why delete the `Default` impl?

Reply:
We were only using it now in the `WorkunitMetadata::new()` associated function. At one point I was pursuing a strategy for solving this problem that was made harder to implement with the Default impl, which is why I removed it. But I think there's no reason to have both a Default impl and a new() pseudo-constructor for the same type.